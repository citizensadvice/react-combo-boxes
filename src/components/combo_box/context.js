import { createContext } from 'react';
import { visuallyHiddenClassName } from '../../constants/visually_hidden_class_name';

export const Context = createContext({
  visuallyHiddenClassName,
  dispatch: null,
});
