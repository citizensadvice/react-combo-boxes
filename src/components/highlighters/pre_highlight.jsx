import React from 'react';
import { HighlightValue } from '../highlight_value';
import { preHighlighter } from '../../highlighters/pre_highlighter';

export function PreHighlight(props) {
  return (
    <HighlightValue
      highlight={preHighlighter}
      {...props}
    />
  );
}
