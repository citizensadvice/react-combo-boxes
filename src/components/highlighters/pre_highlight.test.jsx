import React from 'react';
import { render } from '@testing-library/react';
import { PreHighlight } from './pre_highlight';
import { Context } from '../../context';

function TestHighlight({ children, inverse, ...props }) {
  return (
    <Context.Provider value={{ ...props, props: { value: null } }}>
      <PreHighlight start="<em>" end="</em>" inverse={inverse}>
        {children}
      </PreHighlight>
    </Context.Provider>
  );
}

it('does not highlight with no children', () => {
  const { container } = render((
    <TestHighlight search="bar" />
  ));

  expect(container).toContainHTML('<div></div>');
});

it('does not highlight with no search children', () => {
  const { container } = render((
    <TestHighlight search={null}>
      foo
    </TestHighlight>
  ));

  expect(container).toContainHTML('<div>foo</div>');
});

it('converts <em> in html strings to a highlight', () => {
  const { container } = render((
    <TestHighlight search="bar">
      {'foo <em>bar</em> <em>foe</em>'}
    </TestHighlight>
  ));

  expect(container).toContainHTML('<div>foo <mark>bar</mark> <mark>foe</mark></div>');
});

it('inverses the highlight', () => {
  const { container } = render((
    <TestHighlight inverse search="bar">
      {'foo <em>bar</em> <em>foe</em>'}
    </TestHighlight>
  ));

  expect(container).toContainHTML('<div><mark>foo </mark>bar<mark> </mark>foe</div>');
});
