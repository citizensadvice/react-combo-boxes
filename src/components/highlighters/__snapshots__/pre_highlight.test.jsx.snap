// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts <em> in html strings to a highlight 1`] = `
<div>
  <span
    class="sr-only"
  >
    foo &lt;em&gt;bar&lt;/em&gt; &lt;em&gt;foe&lt;/em&gt;
  </span>
  <span
    aria-hidden="true"
  >
    foo 
    <mark>
      bar
    </mark>
     
    <mark>
      foe
    </mark>
  </span>
</div>
`;

exports[`inverses the highlight 1`] = `
<div>
  <span
    class="sr-only"
  >
    foo &lt;em&gt;bar&lt;/em&gt; &lt;em&gt;foe&lt;/em&gt;
  </span>
  <span
    aria-hidden="true"
  >
    <mark>
      foo 
    </mark>
    bar
    <mark>
       
    </mark>
    foe
  </span>
</div>
`;
